File: app/src/main/java/com/maliotis/petros/stormy/HourAdapter.java
package com.maliotis.petros.stormy;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.maliotis.petros.stormy.weather.Day;
import com.maliotis.petros.stormy.weather.Hour;

public class HourAdapter extends RecyclerView.Adapter<HourAdapter.HourViewHolder> {

    private Hour[] mHours;
    private Context mContext;
    private Day[] mDays;

    public HourAdapter(Context context,Hour[] hours,Day[] days){
        mContext = context;
        mHours = hours;
        mDays = days;
    }

    @Override
    public HourViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.hourly_list_item,parent,false);
        HourViewHolder hourViewHolder = new HourViewHolder(view);
        return hourViewHolder;
    }

    @Override
    public void onBindViewHolder(HourViewHolder holder, int position) {
        holder.bindHour(mHours[position]);

    }

    @Override
    public int getItemCount() {
        return mHours.length;
    }

    public class HourViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        public TextView mTimeLabel;
        public TextView mSummaryLabel;
        public TextView mTemperatureLabel;
        public ImageView mIconImageView;
        public TextView mHourDay;
        public TextView mDegree;


        public HourViewHolder(View itemView) {
            super(itemView);

            mTimeLabel = (TextView) itemView.findViewById(R.id.timeLabel);
            mSummaryLabel  = (TextView) itemView.findViewById(R.id.summaryLabel);
            mTemperatureLabel = (TextView) itemView.findViewById(R.id.temperatureLabel);
            mIconImageView = (ImageView) itemView.findViewById(R.id.iconImageView);
            mHourDay = (TextView) itemView.findViewById(R.id.hourDayLabel);
            mDegree = (TextView) itemView.findViewById(R.id.degreeLabel);


            itemView.setOnClickListener(this);
        }

        public void bindHour(Hour hour){

            if(hour.getHour().equals("0:00")){
                mHourDay.setText("");
                mTimeLabel.setText("");
                mSummaryLabel.setText("");
                mTemperatureLabel.setText("");
                mIconImageView.setImageResource(android.R.color.transparent);
                mHourDay.setText("");
                mDegree.setText("");
                //setting the day of the week.
                mHourDay.setText(hour.getDayOfTheWeekH());
            }
            else {
                mHourDay.setText("");
                mDegree.setText("o");
                mTimeLabel.setText(hour.getHour());
                mSummaryLabel.setText(hour.getSummary());
                mTemperatureLabel.setText(hour.getTemperature() + "");
                mIconImageView.setImageResource(hour.getIconId());
            }
        }

        @Override
        public void onClick(View v) {
            String time = mTimeLabel.getText().toString();
            String temperature = mSummaryLabel.getText().toString();
            String summary = mSummaryLabel.getText().toString();
            String message = String.format("At %s it will be %s and %s",time,temperature,summary);
            Toast.makeText(mContext,message,Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/maliotis/petros/weather/HourAdapter.kt
package com.maliotis.petros.weather

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.maliotis.petros.weather.HourAdapter.HourViewHolder
import com.maliotis.petros.weather.weather.Day
import com.maliotis.petros.weather.weather.Hour

class HourAdapter(private val mContext: Context, private val mHours: Array<Hour>, private val mDays: Array<Day>) : RecyclerView.Adapter<HourViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HourViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.hourly_list_item, parent, false)
        return HourViewHolder(view)
    }

    override fun onBindViewHolder(holder: HourViewHolder, position: Int) {
        holder.bindHour(mHours[position])
    }

    override fun getItemCount(): Int {
        return mHours.size
    }

    inner class HourViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        var mTimeLabel: TextView
        var mSummaryLabel: TextView
        var mTemperatureLabel: TextView
        var mIconImageView: ImageView
        var mHourDay: TextView
        var mDegree: TextView
        fun bindHour(hour: Hour) {
            if (hour.hour == "0:00") {
                mHourDay.text = ""
                mTimeLabel.text = ""
                mSummaryLabel.text = ""
                mTemperatureLabel.text = ""
                mIconImageView.setImageResource(android.R.color.transparent)
                mHourDay.text = ""
                mDegree.text = ""
                //setting the day of the week.
                mHourDay.text = hour.dayOfTheWeekH
            } else {
                mHourDay.text = ""
                mDegree.text = "o"
                mTimeLabel.text = hour.hour
                mSummaryLabel.text = hour.summary
                mTemperatureLabel.text = hour.temperature.toString() + ""
                mIconImageView.setImageResource(hour.iconId)
            }
        }

        override fun onClick(v: View) {
            val time = mTimeLabel.text.toString()
            val temperature = mSummaryLabel.text.toString()
            val summary = mSummaryLabel.text.toString()
            val message = String.format("At %s it will be %s and %s", time, temperature, summary)
            Toast.makeText(mContext, message, Toast.LENGTH_LONG).show()
        }

        init {
            mTimeLabel = itemView.findViewById<View>(R.id.timeLabel) as TextView
            mSummaryLabel = itemView.findViewById<View>(R.id.summaryLabel) as TextView
            mTemperatureLabel = itemView.findViewById<View>(R.id.temperatureLabel) as TextView
            mIconImageView = itemView.findViewById<View>(R.id.iconImageView) as ImageView
            mHourDay = itemView.findViewById<View>(R.id.hourDayLabel) as TextView
            mDegree = itemView.findViewById<View>(R.id.degreeLabel) as TextView
            itemView.setOnClickListener(this)
        }
    }

}